/root/ecg_ai/FocusECG/FocusECG/finetune_test.py:119: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  checkpoint = torch.load(args.ckpt_path, map_location='cpu')
cuda
cuda
Namespace(all_encode_norm_layer='LayerNorm', batch_size=512, ckpt_path='/root/ecg_ai/FocusECG/FocusECG/ckpt/pre_train/icentiallk-p01/FocusMae/202501070232/min_val_loss=23.90247344970703.pth', class_n=5, classifier_head_name='mlp_v1', data_standardization=True, dataset_name='ptb-xl', decoder_depth=8, decoder_embed_dim=64, decoder_num_heads=8, device=device(type='cuda'), early_stop_patience=30, embed_dim=256, encoder_depth=48, encoder_num_heads=16, learning_rate=0.0001, mask_ratio=0.75, mask_type='period', max_epoch_num=1000, mlp_ratio=2, model_name='FocusMae', norm_layer='LayerNorm', num_input_channels=1, num_layers=48, patch_length=50, patch_size=50, patch_stride=50, pretrain_model_freeze=True, scheduler_factor=0.8, scheduler_min_lr=1e-08, scheduler_patience=20, self_attn=True, self_attn_heads=16, signal_length=2500, task='finetune', test_data_path='/root/data/p1-2_people2000_segmentall_sample_step100_data/index_60-80k_step:1.txt', train_data_path='/root/data/ptb-xl/train.txt', use_cls_token=True, use_positional_encoding=True, val_data_path='/root/data/ptb-xl/val.txt', val_every_n_steps=40, weight_decay=0.001)
training:   0%|          | 0/22 [00:00<?, ?it/s]                                                Traceback (most recent call last):
  File "main.py", line 78, in <module>
    run_finetune(args)
  File "/root/ecg_ai/FocusECG/FocusECG/finetune_test.py", line 174, in run_finetune
    pred = model(input_x)
  File "/root/miniconda3/envs/ecg/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/root/miniconda3/envs/ecg/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/ecg_ai/FocusECG/FocusECG/model/classifier.py", line 78, in forward
    embedding = self.pre_train_model.forward_feature(x)
  File "/root/ecg_ai/FocusECG/FocusECG/model/FocusMae.py", line 306, in forward_feature
    x = self.patch_embed(x)
  File "/root/miniconda3/envs/ecg/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/root/miniconda3/envs/ecg/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/root/ecg_ai/FocusECG/FocusECG/model/FocusMae.py", line 73, in forward
    B, C, L = x.shape
ValueError: not enough values to unpack (expected 3, got 2)
